### PR TITLE
Fix #401: Fix traceback and inaccessible index by ignoring removed bases.

### LINKED ISSUE #401: AttributeError: 'NoneType' object has no attribute 'ixconfig'
- Bitbucket: https://bitbucket.org/hpk42/devpi/issue/401
- Originally reported by: @tomster
- Originally created at: 2017-03-20T18:16:59.943

when accessing `https://pypi.senic.com/tomster/master/+simple/` i get the following traceback.

```
2017-03-20 17:14:02,927 ERROR Exception when serving /tomster/master/+simple/
Traceback (most recent call last):
  File "/home/devpi/devpi/lib/python2.7/site-packages/waitress/channel.py", line 338, in service
    task.service()
  File "/home/devpi/devpi/lib/python2.7/site-packages/waitress/task.py", line 169, in service
    self.execute()
  File "/home/devpi/devpi/lib/python2.7/site-packages/waitress/task.py", line 399, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/home/devpi/devpi/lib/python2.7/site-packages/devpi_server/views.py", line 146, in __call__
    return self.app(environ, start_response)
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/router.py", line 233, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/router.py", line 208, in invoke_subrequest
    response = handle_request(request)
  File "/home/devpi/devpi/lib/python2.7/site-packages/devpi_server/views.py", line 169, in request_log_handler
    response = handler(request)
  File "/home/devpi/devpi/lib/python2.7/site-packages/devpi_server/views.py", line 204, in request_tx_handler
    response = handler(request)
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/tweens.py", line 62, in excview_tween
    reraise(*attrs['exc_info'])
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/tweens.py", line 22, in excview_tween
    response = handler(request)
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/router.py", line 155, in handle_request
    view_name
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/view.py", line 612, in _call_view
    response = view_callable(context, request)
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/viewderivers.py", line 409, in viewresult_to_response
    result = view(context, request)
  File "/home/devpi/devpi/lib/python2.7/site-packages/pyramid/viewderivers.py", line 123, in _class_requestonly_view
    response = getattr(inst, attr)()
  File "/home/devpi/devpi/lib/python2.7/site-packages/devpi_server/views.py", line 470, in simple_list_all
    stage_results = list(stage.op_sro("list_projects_perstage"))
  File "/home/devpi/devpi/lib/python2.7/site-packages/devpi_server/model.py", line 514, in op_sro
    for stage in self.sro():
  File "/home/devpi/devpi/lib/python2.7/site-packages/devpi_server/model.py", line 558, in sro
    if current_stage.ixconfig['type'] == 'mirror':
AttributeError: 'NoneType' object has no attribute 'ixconfig'
```

i'm running the latest versions:

```
      devpi-common==3.0.1
      devpi-server==4.2.1
      devpi-web==3.1.1
```

i'm not sure when exactly this behaviour started...

### PR DESCRIPTION
**Pull request by fschulze**
See #401 and #142

### CODE REVIEW

File: server/devpi_server/model.py [hunk 1]
```diff
@@ -560,6 +560,8 @@ def sro(self):
             seen.add(stage.name)
             for base in stage.ixconfig.get("bases", ()):
                 current_stage = self.model.getstage(base)
+                if current_stage is None:
+                    continue
```
Comment (hpk42) [inline server/devpi_server/model.py]: shouldn't this do a warning so that at least something is in the server logs?

**Comment by hpk42** [comment]
i am kind of ok with the "workaround fix" but in fact we should do more when removing an index -- it's not a very common operation i think so crawling through all indexes and removing this index also from bases (maybe with a special option) seems like the proper way, no?  (we can still merge this PR i guess for now -- just wanted to reconcile  our undestandings)

**Comment by fschulze** [comment]
I wanted to fix inaccessible indices first. I will make a separate PR for anything that goes further.

File: server/devpi_server/model.py (hunk 1 above)
Comment (fschulze) [inline server/devpi_server/model.py]: Agreed. Will add.

**Comment by fschulze** [comment]
I've added the logging.

**Pull request merged**